### DISCUSSION TITLE
How to get current autoplay time?

### DISCUSSION THREAD

**Discussion by kitaliev** [nolimits4web/swiper#5884] [Q&A] [answered]
I have an external progress bar that indicates time until next slide will appear and i wanted to get current autoplay time and connect that values.
didn't find any related events/props/methods in docs or issues/discussions in github so decided to ask
I could make an independent interval/timeout but it didn't seems like right way

**Comment by Arhell** [comment, marked as answer]
Try
https://codepen.io/Bluxart/pen/MWyejBr
https://codepen.io/bjlashbrook/pen/byXzdB

**Reply by kitaliev** [reply]
That is indeed what i wanted
But i ended up doing other solution
I disabled swiper library autoplay and used progressbar.js library as my progress bar, it provides a callback in `animate` method which i used to call swiper's `slideNext` method
also it provides usefull methods like `stop` and `set`, since i use progressbar.js as my "timer" i need to stop and start progress again when i hover

**Reply by Arhell** [reply]
Cool, also a good solution.
You can always find, make the decision you need
